wlauto/workloads/msword/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,376 @@
+        // signIn();
+        startLogger("welcome_screen_skip");
+        clickUiObject(BY_TEXT, "Skip", true); // skip welcome screen
+        stopLogger("welcome_screen_skip");
+        if ("create".equalsIgnoreCase(parameters.getString("test_type"))) {
+            testCreateDocument();
+        } else {
+            testExistingDocument();
+        }
+        unsetScreenOrientation();
+        writeResultsToFile(results, parameters.getString("output_file"));
+    }
+
+    protected void signIn() throws Exception {
+        clickUiObject(BY_TEXT, "Sign in", true);
+        UiObject emailField = new UiObject(new UiSelector().className(CLASS_EDIT_TEXT));
+        emailField.clearTextField();
+        emailField.setText(loginEmail + "@"); // deliberately incorrect
+        clickUiObject(BY_DESC, "Next", CLASS_BUTTON, true);
+        waitForProgress(WAIT_TIMEOUT_5SEC);
+
+        // When login email is not valid, app is redirected to another screen
+        // to choose between Microsoft account or work-provided account
+        // Full text says: We're having trouble locating your account.
+        UiObject troublePage = new UiObject(new UiSelector().descriptionContains("trouble locating"));
+        if (!troublePage.exists()) {
+            throw new UiObjectNotFoundException("Not found: Login problem page");
+        }
+        clickUiObject(BY_DESC, "Work account", CLASS_BUTTON, true);
+        waitForProgress(WAIT_TIMEOUT_5SEC);
+        // Here the views inside the WebView can now be accessed as normal
+        UiObject webview = new UiObject(
+            new UiSelector().className("android.webkit.WebView").descriptionContains("Sign in"));
+        if (!webview.waitForExists(WAIT_TIMEOUT_5SEC)) {
+            throw new UiObjectNotFoundException("Not found: Sign-in WebView");
+        }
+        emailField = new UiObject(new UiSelector().className(CLASS_EDIT_TEXT).instance(0));
+        int textLength = 30;
+        emailField.clickBottomRight();
+        while (textLength > 0) {
+            getUiDevice().pressDelete();
+        }
+        getUiDevice().waitForIdle();
+        emailField.click();
+        emailField.setText(loginEmail);
+        UiObject passwordField = new UiObject(new UiSelector().className(CLASS_EDIT_TEXT).instance(1));
+        passwordField.setText(loginPass);
+        clickUiObject(BY_DESC, "Sign in", CLASS_BUTTON, true);
+    }
+
+    public void testCreateDocument() throws Exception {
+        newDocument("Newsletter");
+        // Dismiss tooltip if it appears
+        UiObject tooltip = new UiObject(new UiSelector().textContains("Got it").className(CLASS_BUTTON));
+        startLogger("document_tooltip");
+        if (tooltip.waitForExists(WAIT_TIMEOUT_5SEC)) {
+            tooltip.click();
+        }
+        stopLogger("document_tooltip");
+
+        // show command palette
+        UiObject paletteToggle = new UiObject(new UiSelector().resourceId(packageID + "paletteToggleButton"));
```

Comment:
For consistency why not put this before the click action like you have elsewhere.